### ISSUE TITLE
Use of uninitialized value $inode in numeric ne (!=) at /opt/perl5/perlbrew/perls/perl-5.36.0/lib/site_perl/5.36.0/Log/Dispatch/FileRotate/Mutex.pm line 80.

### ISSUE THREAD

**Issue by davehodg** [mschout/perl-log-dispatch-filerotate#29]
We're getting this error at a certain time of the night, like 02:35 (occasionally 02:48 and 02:55). Looking at the surrounding code:

    elsif ($self->{$pid} == 0) {
        # file is open, but not locked.
        if (safe_flock($self->{_fh}, LOCK_EX)) {
            my ($inode) = (stat $self->{_path})[1];

    (80:)    if ($inode != $self->{_inode}) {
                # file was removed or changed underneath us, reopen instead
                delete $self->{$pid};

                close(delete $self->{_fh});

                delete $self->{$pid};
                delete $self->{_inode};

                return $self->lock;
            }

            $self->{$pid} = 1;
        }

Why would $inode be null? Also, the return from safe_flock isn't checked so the log file may have been rotated away.

**Comment by mschout** [comment]
Good catch @davehodg  if the file was rotated away, the result of stat is going to be nothing, so that's likely why.
